ocaml/xapi/xapi_host.ml

```diff
@@ -67,19 +67,33 @@ let set_power_on_mode ~__context ~self ~power_on_mode ~power_on_config =
   Xapi_host_helpers.update_allowed_operations ~__context ~self
 
 (** Before we re-enable this host we make sure it's safe to do so. It isn't if:
-    	+ we're in the middle of an HA shutdown/reboot and have our fencing temporarily disabled.
-    	+ xapi hasn't properly started up yet.
-    	+ HA is enabled and this host has broken storage or networking which would cause protected VMs
-    	to become non-agile
+        + there are pending mandatory guidances on the host
+        + we're in the middle of an HA shutdown/reboot and have our fencing temporarily disabled.
+        + xapi hasn't properly started up yet.
+        + HA is enabled and this host has broken storage or networking which would cause protected VMs
+        to become non-agile
 *)
 let assert_safe_to_reenable ~__context ~self =
   assert_startup_complete () ;
+  let host_pending_mandatory_guidances =
+    Db.Host.get_pending_guidances ~__context ~self
+  in
+  if host_pending_mandatory_guidances <> [] then
+    raise
```

Comment:
Maybe:
```
info "%s: %d mandatory guidance is pending for %s" __FUNCTION__ (List.length host_pending_mandatory_guidances) (Ref.string_of self)
```

Reply:
Yes. The printing actions in mandatory guidance into the log seems not necessary.